conans/test/functional/options/options_test.py

```diff
@@ -33,37 +29,29 @@ def test(self):
         client.run("create . Pkg/0.1@user/testing -o Pkg:shared=2")
         self.assertIn("Pkg/0.1@user/testing: BUILD SHARED: 2", client.out)
         client.run("create . Pkg/0.1@user/testing -o shared=1", assert_error=True)
-        self.assertIn("'options.shared' doesn't exist", client.out)
+        self.assertIn("option 'shared' doesn't exist", client.out)
 
-        conanfile = """from conans import ConanFile
-class Pkg(ConanFile):
-    pass
-"""
-        client.save({"conanfile.py": conanfile}, clean_first=True)
+        client.save({"conanfile.py": GenConanfile()}, clean_first=True)
         client.run("create . Pkg/0.1@user/testing -o *:shared=True")
         self.assertIn("Pkg/0.1@user/testing: Calling build()", client.out)
         client.run("create . Pkg/0.1@user/testing -o shared=False", assert_error=True)
-        self.assertIn("'options.shared' doesn't exist", client.out)
+        self.assertIn("option 'shared' doesn't exist", client.out)
         # With test_package
         client.save({"conanfile.py": conanfile,
                      "test_package/conanfile.py": test})
-        client.run("create . Pkg/0.1@user/testing -o *:shared=True")
-        self.assertIn("Pkg/0.1@user/testing: Calling build()", client.out)
-        self.assertIn("Pkg/0.1@user/testing (test package): Calling build()", client.out)
+        client.run("create . Pkg/0.1@user/testing -o *:shared=True", assert_error=True)
+        self.assertIn("ERROR: Pkg/0.1@user/testing: 'True' is not a valid 'options.shared' value",
+                      client.out)
```

Comment:
This test is checking that `shared` in `conan create . Pkg/0.1@user/testing -o shared=False` refers to a specific package and raises because the option doesn't exist; but with `conan create . Pkg/0.1@user/testing -o *:shared=True` it doesn't raise because it is not specific to any package.

Reply:
It tries to assign a ``True`` value to an option that only has ``[1, 2]`` as possible values. I find it reasonable to fail, isn't it?

Reply:
oh! I see, you have changed the `conanfile` itself. Before it was:

```
conanfile = """from conans import ConanFile
class Pkg(ConanFile):
    pass
"""
```

now it is a different conanfile. I would keep the same behavior in this test, it is just about writing a `GenConanfile()`

Reply:
Nevermind, I think there is an error, let me check it.

Reply:
Yeah, an annoying bug because of using the same test() to test 2 different things. I have splitted the test in 2.